staker-rewards/lib/assets-calculator.js

```diff
@@ -0,0 +1,182 @@
+import { callWithRetry, getPastEvents } from "./contract-helper.js"
+
+export default class AssetsCalculator {
+  constructor(
+    context,
+    interval,
+    tokenStaking,
+    keepRandomBeaconOperator,
+    bondedECDSAKeepFactory,
+    keepBonding,
+    sortitionPoolAddress
+  ) {
+    this.context = context
+    this.interval = interval
+    this.tokenStaking = tokenStaking
+    this.keepRandomBeaconOperator = keepRandomBeaconOperator
+    this.bondedECDSAKeepFactory = bondedECDSAKeepFactory
+    this.keepBonding = keepBonding
+    this.sortitionPoolAddress = sortitionPoolAddress
+  }
+
+  static async initialize(context, interval) {
+    const { contracts } = context
+
+    const bondedECDSAKeepFactory = await contracts.BondedECDSAKeepFactory.deployed()
+    const tbtcSystem = await contracts.TBTCSystem.deployed()
```

Comment:
Instead of importing `tbtc` as a dependency and calling `TBTCSystem.deployed()`, I'd hardcode `TBTCSystem` address as a `SanctionedApplication` constant in `rewards.js`.

Ideally, `keep-ecdsa/staker-rewards` shouldn't need to know anything about applications other than the client's config file and `staker-rewards` should be application-agnostic. All `staker-rewards` scripts should now are what the sanctioned applications are.

Reply:
Yep good idea. However, I prefer putting the `SanctionedApplication` in the `context.js` file to keep it close to other ABIs imports and avoid importing `reward.js` in `assets-calculator.js`.